pennylane/_device.py

```diff
@@ -406,6 +407,11 @@ def execute(self, queue, observables, parameters={}, **kwargs):
         self._parameters.update(parameters)
 
         results = []
+        if self._shot_vector is not None:
+            warnings.warn(
+                "Specifying a list of shots is only supported for"
+                "QubitDevice based devices. Falling back to executions using all shots in the shot list."
+            )
```

Comment:
Does this rely on QubitDevice never calling `super().execute(...)`? Could be worth adding a comment about that, so that it's clear why this doesn't affect _all_ devices.

Reply:
Good point! Yep, for `QubitDevice` the `execute` method differs significantly from `Device.execute`. Let me add a comment.